VultisigApp/VultisigApp/Services/Ton/TonService.swift

```diff
@@ -0,0 +1,119 @@
+import Foundation
+
+// Define the structures for responses and interfaces
+struct TonAddressInformation: Codable {
+    var balance: String
+    var code: String
+    var data: String
+    var frozen_hash: String
+    var last_transaction_hash: String
+    var last_transaction_lt: String
+    var status: String
+}
+
+struct ApiResponse<T: Codable>: Codable {
+    var ok: Bool
+    var result: T?
+}
+
+struct ResultData: Codable {
+    struct AddressInfo: Codable {
+        let type = "accountAddress"
+        let account_address: String
+    }
+
+    struct LastTransactionIdInfo: Codable {
+        let type = "internal.transactionId"
+        let lt: String
+        let hash: String
+    }
+
+    struct BlockIdInfo: Codable {
+        let type = "ton.blockIdExt"
+        let workchain: Int
+        let shard: String
+        let seqno: Int
+        let root_hash: String
+        let file_hash: String
+    }
+
+    struct AccountStateInfo: Codable {
+        let type = "wallet.v4.accountState"
+        let wallet_id: String
+        let seqno: Int
+    }
+
+    var type = "fullAccountState"
+    var address: AddressInfo
+    var balance: String
+    var last_transaction_id: LastTransactionIdInfo
+    var block_id: BlockIdInfo
+    var sync_utime: Int
+    var account_state: AccountStateInfo
+    var revision: Int
+    let extra: String
+}
+
+class TonService {
+    
+    static let shared = TonService()
+
+    func broadcastTransaction(_ obj: String) async throws -> String {
+                
+        let body: [String: Any] = ["boc": obj]
+        let dataPayload = try JSONSerialization.data(withJSONObject: body, options: [])
+        let data = try await Utils.asyncPostRequest(urlString: "https://toncenter.com/api/v2/sendBocReturnHash", headers: [:], body: dataPayload)
```

Comment:
can you please move this URL to `Endpoint` as well

Reply:
Oh, I forgot to push it.